### PR TITLE
add todo percentage in item list

### PR DESCRIPTION
**Pull request by XGHeaven**
#750 

And I implement it like github todo list.

Before:
qq20170810-123435

After:
qq20170810-123357
qq20170810-123338

Color as same as TodoListPercentage.
Todo style is like github todo style. And todo list doesn't show in folded mode.

Added:
* browser/lib/getTodoState.js get percentage of complete todo helper function
* browser/compoents/TodoProcess.{js/styl}

### CODE REVIEW

**Comment by kazup01** [comment]
Hi @XGHeaven , thanks for your PR!
How about like GitHub todo list? I think that this UI is clearer.

screen shot 0029-08-09 at 11 54 05 am

**Comment by XGHeaven** [comment]
@BoostnoteBot Why need to change `browser/main/Detail/SnippetNoteDetail.js`?
The todo percentage only show when note type is `MARKDOWN_NOTE`, snippet note haven't todo percentage.

**Comment by asmsuechan** [comment]
@XGHeaven No worry, it's a bot. So it replays automatically when `MarkdownNoteDetail` is detected.

**Comment by asmsuechan** [comment]
I don't think it should be shown in folded note list.

File: browser/lib/getPercentage.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export default function getPercentage(content) {
```
Comment (asmsuechan) [inline browser/lib/getPercentage.js]: Great!

**Comment by XGHeaven** [comment]
@kazup01 I think this is a good idea. 
@asmsuechan What are you think about that? use github todo list?
If percentage don't shown in folded list, user want to see todo percentage only switch to unfolded list. It's a little trouble.

**Comment by asmsuechan** [comment]
I think you don't have to implement the GitHub like todo list in this time. Perhaps I'll implement it later on.

I want the folded list to be simple. And come to think of the updated time of a note, it's not be shown on folded list.

**Comment by XGHeaven** [comment]
I think github todo list looks like not bad.
image
If you implement it, that's good.

**Comment by asmsuechan** [comment]
> I think github todo list looks like not bad.

I think so, but I have another idea the reason why I mentioned `you don't have to implement the GitHub like todo list in this time`.
I also want to put the numbers of todo percentage in `MarkdownNoteDetail`. And it's not only this item list's problem. So I want to separate them into other PRs.

**Comment by asmsuechan** [comment]
Ah, of course, you can do it, please do it.

**Comment by XGHeaven** [comment]
Ok

**Comment by XGHeaven** [comment]
@asmsuechan I implement it. Can you review it?

File: browser/lib/getTodoState.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export function getTodoState (content) {
```
Comment (asmsuechan) [inline browser/lib/getTodoState.js]: I prefer `getTodoState` than `getTodoStatus`.

Comment (XGHeaven) [inline browser/lib/getTodoState.js]: How do you think about `getTodoState.js` file name? Should change to `getTodoStatus.js`?

Comment (asmsuechan) [inline browser/lib/getTodoState.js]: I want to change both of them.

Comment (XGHeaven) [inline browser/lib/getTodoState.js]: I changed that. And I also changed TodoProcess prop from `todoState` to `todoStatus`.
By the way, add transition for TodoProcess bar.

**Comment by asmsuechan** [comment]
@XGHeaven  Well done! Thank you so much indeed :blush: I'll merge this later.

**Comment by XGHeaven** [comment]
😄

**Pull request merged**